lib/ngx/re.md

```diff
@@ -0,0 +1,182 @@
+Name
+====
+
+ngx.re - Lua API for convenience utilities for `ngx.re`.
+
+Table of Contents
+=================
+
+* [Name](#name)
+* [Status](#status)
+* [Synopsis](#synopsis)
+* [Description](#description)
+* [Methods](#methods)
+    * [split](#split)
+* [Community](#community)
+    * [English Mailing List](#english-mailing-list)
+    * [Chinese Mailing List](#chinese-mailing-list)
+* [Bugs and Patches](#bugs-and-patches)
+* [Author](#author)
+* [Copyright and License](#copyright-and-license)
+* [See Also](#see-also)
+
+Status
+======
+
+This Lua module is currently considered experimental.
+
+Synopsis
+========
+
+```lua
+local ngx_re = require "ngx.re"
+
+-- split
+local res, err = ngx_re.split("a,b,c,d", ",")
+--> res is now {"a", "b", "c", "d"}
+```
+
+[Back to TOC](#table-of-contents)
+
+Description
+===========
+
+This Lua module provides a Lua API which implements convenience utilities for
+the `ngx.re` API.
+
+[Back to TOC](#table-of-contents)
+
+Methods
+=======
+
+All the methods of this module are static (or module-level). That is, you do
+not need an object (or instance) to call these methods.
+
+[Back to TOC](#table-of-contents)
+
+split
+----------------
+**syntax:** *res, err = ngx_re.split(subject, regex, options?, ctx?, max?, res?)*
+
+Splits the `subject` string using the Perl compatible regular expression
+`regex` with the optional `options`.
+
+This function returns a Lua (array) table (with integer keys) containing the
+splitted values.
+
+In case of error, `nil` will be returned as well as a string describing the
+error.
+
+When `regex` contains a sub-match capturing group, and when such a match is
+found, the capture will be inserted in between each splitted value, like so:
```

Comment:
"the capture will be inserted" => "the *first* submatch capture  will be inserted".